Jenkinsfile

```diff
@@ -546,6 +546,15 @@ pipeline {
             }
           }
         }
+        stage('Publish Docker image') {
```

Comment:
Question for Jenkins experts:

This is in the `Release` stage, because we only want to publish the image, when we do a release.

But the building of the image happens earlier in the `Create Docker image` stage which run after our tests on each build - but that's outside of the `Release` stage. The idea here is that we always build the image (to make sure we don't break something, but only publish when we release).

Question: will the created docker image, which we create in the `Create Docker image` stage visible here?

Reply:
That's correct, all the stages are using the top-level agent one defined after the `pipeline {`, unless they are override with `agent { label `, and I see it's not the case unless running for Windows or the Benchmarks.

So, it's visible